pkg/resource/notebook_instance/hooks.go

```diff
@@ -0,0 +1,49 @@
+package notebook_instance
+
+import (
+	"context"
+	"errors"
+
+	ackrequeue "github.com/aws-controllers-k8s/runtime/pkg/requeue"
+	svccommon "github.com/aws-controllers-k8s/sagemaker-controller/pkg/common"
+	svcsdk "github.com/aws/aws-sdk-go/service/sagemaker"
+)
+
+var (
+	modifyingStatuses = []string{
+		svcsdk.NotebookInstanceStatusPending,
+		svcsdk.NotebookInstanceStatusUpdating,
+		svcsdk.NotebookInstanceStatusDeleting,
+		svcsdk.NotebookInstanceStatusStopping,
+	}
+
+	resourceName = resourceGK.Kind
+
+	requeueWaitWhileDeleting = ackrequeue.NeededAfter(
+		errors.New(resourceName+" is deleting."),
+		ackrequeue.DefaultRequeueAfterDuration,
+	)
+	requeueWaitWhileStopping = ackrequeue.NeededAfter(
+		errors.New(resourceName+" is stopping."),
+		ackrequeue.DefaultRequeueAfterDuration,
+	)
+)
+
+// requeueUntilCanModify creates and returns an
+// ackrequeue error if a resource's latest status matches
+// any of the defined modifying statuses below.
+func (rm *resourceManager) requeueUntilCanModify(
+	ctx context.Context,
+	r *resource,
+) error {
+	latestStatus := r.ko.Status.NotebookInstanceStatus
+	return svccommon.RequeueIfModifying(latestStatus, &resourceName, &modifyingStatuses)
+}
+
+// customSetOutput sets the ack syncedCondition depending on
+// whether the latest status of the resource is one of the
+// defined modifyingStatuses.
+func (rm *resourceManager) customSetOutput(r *resource) {
```

Comment:
this hook is in a file called custom_set_output.go for other resources, any reason it is called hooks here ?

Reply:
It was a suggestion. Feature Group/Model package use the same format.
https://github.com/aws-controllers-k8s/sagemaker-controller/pull/56#discussion_r685611827

Reply:
amount of code has reduced, thanks for svccommon. So we thought it's ok to consolidate hook related code